app/src/main/java/com/cornellappdev/uplift/ui/screens/ProfileCreationScreen.kt

```diff
@@ -0,0 +1,227 @@
+import com.cornellappdev.uplift.util.montserratFamily
+import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import com.google.accompanist.systemuicontroller.SystemUiController
+import com.google.accompanist.systemuicontroller.rememberSystemUiController
+
+/**
+ * The profile creation page during the Uplift onboarding process.
+ */
+@OptIn(ExperimentalPermissionsApi::class, ExperimentalMaterial3Api::class)
+@Composable
+fun ProfileCreationScreen() {
+    val systemUiController: SystemUiController = rememberSystemUiController()
+    val checkboxColors: CheckboxColors =
+        CheckboxDefaults.colors(checkedColor = PRIMARY_YELLOW, checkmarkColor = Color.Black, uncheckedColor = GRAY03)
+
+    systemUiController.isStatusBarVisible = false
+    systemUiController.isNavigationBarVisible = false // Navigation bar
+    systemUiController.isSystemBarsVisible = false
+
+    Scaffold(topBar = {
+        TopAppBar(
+            title = {
+                Text(
+                    text = "Complete your profile.",
+                    modifier = Modifier.padding(vertical = 4.dp),
+                    fontSize = 24.sp,
+                    fontFamily = montserratFamily,
+                    fontWeight = FontWeight.Bold
+                )
+            },
+            modifier = Modifier.shadow(elevation = 20.dp, ambientColor = GRAY01)
+        )
+    }) { innerPadding ->
+        val checkedState = remember { mutableStateOf(false) }
+        val checkedState2 = remember { mutableStateOf(false) }
+        val checkedState3 = remember { mutableStateOf(false) }
+
+        Column(
+            modifier = Modifier
+                .padding((innerPadding))
+                .fillMaxWidth(),
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+            Spacer(modifier = Modifier.height(46.dp))
+            //TODO - this Box placeholder will be replaced with a component to choose photo
+            Box(
+                modifier = Modifier
+                    .size(155.dp)
+                    .shadow(
+                        elevation = 20.dp,
+                        shape = CircleShape,
+                        ambientColor = GRAY01,
+                        spotColor = GRAY03
+                    )
+                    .background(color = GRAY01, shape = CircleShape)
+                    .border(width = 8.dp, color = Color.White, shape = CircleShape),
+                contentAlignment = Alignment.Center
+            ) {
+                Image(
+                    painter = painterResource(R.drawable.camera),
+                    contentDescription = "",
+                    modifier = Modifier.offset(y = 5.dp)
+                )
+            }
+            Text(
+                text = "First name Last name",
+                modifier = Modifier.padding(vertical = 24.dp),
+                fontSize = 24.sp,
+                fontFamily = montserratFamily,
+                fontWeight = FontWeight.Bold
+            )
```

Comment:
Not necessarily something that needs a code change, but how are you planning to implement this with ViewModels? This for example is info that we'd need to get from the VM layer somehow.

Are you planning to have a sort of "Root" onboarding screen with ONLY ONE overarching viewmodel? Or one different viewmodel for each onboarding screen?

I believe on Resell I'm doing one viewmodel per screen, but it's arguably excessive.

Reply:
I assume I would have one view model for the entire onboarding flow because the only interaction with the data is getting their name from the initial sign in to display instead of "First name Last name" and then saving their profile picture/name changes to the view model also.